torchprime/torch_xla_models/mixtral/model.py

```diff
@@ -358,13 +359,12 @@ def forward(self, dispatch_input):
     mesh = xs.get_global_mesh()
     assert mesh is not None
     layer_w1 = torch.einsum("ebcm,emh->ebch", dispatch_input, self.w1)
-    xs.mark_sharding(layer_w1, mesh, ("expert", ("data", "fsdp"), None, None))
+    MarkShardingFunction.apply(layer_w1, mesh, ("expert", ("data", "fsdp"), None, None))
```

Comment:
If `MarkShardingFunction` is no longer in-place as of https://github.com/pytorch/xla/pull/8821, wouldn't we need to write it like

```
layer_w1 = MarkShardingFunction.apply(layer_w1, mesh, ("expert", ("data", "fsdp"), None, None))
```

If we don't use the return value of `MarkShardingFunction`, I suspect `layer_w1` won't be sharded at all. The E2E test probably passed because we're not checking memory usage and it happened to fit on a v6e-4. Do you think that could be the case?